plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/BaseDeltaLakeConnectorSmokeTest.java

```diff
@@ -1746,6 +1750,97 @@ public void testVacuum()
+            // operations should be logged
+            assertThat(query("SELECT version, operation, MAP_FILTER(operation_parameters, (k, v) -> k <> 'queryId') FROM \"" + tableName + "$history\"")).matches( """
+                    VALUES\s
+                    (CAST(0 AS BIGINT), CAST('CREATE TABLE AS SELECT' AS VARCHAR), CAST(MAP() AS MAP(VARCHAR, VARCHAR))),
+                    (1, 'MERGE', MAP()),
+                    (2, 'VACUUM START', MAP()),
+                    (3, 'VACUUM END', MAP(ARRAY['status'], ARRAY['COMPLETED'])),
+                    (4, 'VACUUM START', MAP()),
+                    (5, 'VACUUM END', MAP(ARRAY['status'], ARRAY['COMPLETED']))""");
+            assertThat(Integer.parseInt((String) computeScalar("SELECT operation_metrics['sizeOfDataToDelete'] FROM \"" + tableName + "$history\" WHERE version = 2"))).isEqualTo(0);
+            assertThat(Integer.parseInt((String) computeScalar("SELECT operation_metrics['numFilesToDelete'] FROM \"" + tableName + "$history\" WHERE version = 2"))).isEqualTo(0);
+            assertThat(Integer.parseInt((String) computeScalar("SELECT operation_metrics['numDeletedFiles'] FROM \"" + tableName + "$history\" WHERE version = 3"))).isEqualTo(0);
+            assertThat(Integer.parseInt((String) computeScalar("SELECT operation_metrics['numVacuumedDirectories'] FROM \"" + tableName + "$history\" WHERE version = 3"))).isEqualTo(8);
+            assertThat(Integer.parseInt((String) computeScalar("SELECT operation_metrics['sizeOfDataToDelete'] FROM \"" + tableName + "$history\" WHERE version = 4"))).isGreaterThanOrEqualTo(5012);
+            assertThat(Integer.parseInt((String) computeScalar("SELECT operation_metrics['numFilesToDelete'] FROM \"" + tableName + "$history\" WHERE version = 4"))).isEqualTo(5);
+            assertThat(Integer.parseInt((String) computeScalar("SELECT operation_metrics['numDeletedFiles'] FROM \"" + tableName + "$history\" WHERE version = 5"))).isEqualTo(5);
+            assertThat(Integer.parseInt((String) computeScalar("SELECT operation_metrics['numVacuumedDirectories'] FROM \"" + tableName + "$history\" WHERE version = 5"))).isEqualTo(8);
+
+            assertUpdate(sessionWithVacuumLoggingDisabled, "CALL system.vacuum(schema_name => CURRENT_SCHEMA, table_name => '" + tableName + "', retention => '1s')");
+            // no new vacuum logging operations are logged
+            assertQuery("SELECT version, operation FROM \"" + tableName + "$history\"", """
+                    VALUES
+                    (0, 'CREATE TABLE AS SELECT'),
+                    (1, 'MERGE'),
+                    (2, 'VACUUM START'),
+                    (3, 'VACUUM END'),
+                    (4, 'VACUUM START'),
+                    (5, 'VACUUM END')""");
+        }
+        finally {
+            assertUpdate("DROP TABLE " + tableName);
+        }
+    }
+
+    @Test
+    public void testVacuumWithParquetTransactionLogging()
```

Comment:
Could you make `testVacuum` parametrized with `checkpoint_interval` as both test are almost the same?